packages/commonwealth/server/routes/feed.ts

```diff
@@ -0,0 +1,136 @@
+import { AppError } from '@hicommonwealth/core';
+import { Thread, ThreadAttributes, type DB } from '@hicommonwealth/model';
+import { Feed } from 'feed';
+import { slugify } from 'utils';
+import { GetBulkThreadsResult } from '../controllers/server_threads_methods/get_bulk_threads';
+import { ServerControllers } from '../routing/router';
+import { TypedRequestQuery, TypedResponse } from '../types';
+import { formatErrorPretty } from '../util/errorFormat';
+import {
+  ActiveThreadsRequestQuery,
+  BulkThreadsRequestQuery,
+  GetThreadsRequestQuery,
+  GetThreadsResponse,
+  SearchThreadsRequestQuery,
+} from './threads/get_threads_handler';
+
+function toDate(t: ThreadAttributes): Date {
+  return t.last_edited ?? t.created_at;
+}
+
+function sorter(a: ThreadAttributes, b: ThreadAttributes) {
+  return toDate(a).getTime() - toDate(b).getTime();
+}
+
+function computeUpdated(bulkThreads: GetBulkThreadsResult): Date {
+  if (bulkThreads.threads.length === 0) {
+    // there are no threads
+    return new Date();
+  }
+
+  const sorted = [...bulkThreads.threads].sort(sorter);
+
+  // return the most recent thread and get its date
+  return toDate(sorted.reverse()[0]);
+}
+export const getFeedHandler = async (
+  models: DB,
+  controllers: ServerControllers,
+  req: TypedRequestQuery<
+    GetThreadsRequestQuery &
+      (
+        | ActiveThreadsRequestQuery
+        | SearchThreadsRequestQuery
+        | BulkThreadsRequestQuery
+      )
+  >,
+  res: TypedResponse<GetThreadsResponse>,
+) => {
+  const queryValidationResult = Thread.GetThreadsParamsSchema.safeParse(
+    req.query,
+  );
+
+  if (queryValidationResult.success === false) {
+    throw new AppError(formatErrorPretty(queryValidationResult));
+  }
+
+  const { thread_ids, bulk, active, search, community_id } =
+    queryValidationResult.data;
+
+  if (active || search || thread_ids) {
+    throw new Error('Not implemented');
+  }
+
+  // get bulk threads
+  if (bulk) {
+    const bulkQueryValidationResult =
+      Thread.GetBulkThreadsParamsSchema.safeParse(req.query);
+
+    if (bulkQueryValidationResult.success === false) {
+      throw new AppError(formatErrorPretty(bulkQueryValidationResult));
+    }
+
+    const {
+      stage,
+      topic_id,
+      includePinnedThreads,
+      limit,
+      page,
+      orderBy,
+      from_date,
+      to_date,
+      archived,
+    } = bulkQueryValidationResult.data;
+
+    const bulkThreads = await controllers.threads.getBulkThreads({
```

Comment:
I am generally more of a fan of vertical slice type logic. Because in this case if we change getBulkThreads to change the getBulkThreads api, then it can break this API. But I think it is fine for now.

A better approach could be to refactor out the orderBy logic into a separate helper function, which you can then use to construct a sql query to get the exact data you want. This has the benefits of:
1. You don't throw away a bunch of data pulled from the database (We are only using the thread body and date).
2. We will only need to make one db request, and won't need to make a separate one for communities

But I think this approach is fine for now since it is already done and it works.

Reply:
> I am generally more of a fan of vertical slice type logic. Because in this case if we change getBulkThreads to change the getBulkThreads api, then it can break this API. But I think it is fine for now.

That was the goal actually.  This code had to duplicate some functionality in the previous 'threads' call but they're also not 100% the same.  Since this is a 'feed' view I needed a way for them to share the logic but without refactoring the original.

But yeah. I don't disagree with your thoughts but this was also my first PR and didn't want to really refactor a tons of stuff!